### ISSUE TITLE
Privacy Badger has stopped blocking anything

### ISSUE THREAD

**Issue by joepie91** [EFForg/privacybadgerchrome#353]
As far as I can tell, Privacy Badger has stopped functioning completely. Every single domain shows as green/allowed now - even obviously bad things like b.scorecardresearch.com and embedded Facebook comments - and all the content is being loaded, so it is evidently not just a display bug.

I have restarted my browser, disabled/enabled Privacy Badger, and even reinstalled it, but it makes no difference.

* Operating system: openSUSE 13.1 x86_64
* Chrome version: 41.0.2272.118 (64-bit)
* PB version: 2015.4.1

I have a lot of other extensions installed - is there some way to get a copy-pastable list of them?

**Comment by cooperq** [comment]
you can go to chrome://extensions that will show you the list of extensions you have installed. Was privacy badger functioning before? Are you saying that domains which were previously blocked are now unblocked?

**Comment by burgwinkel** [comment]
Same here. Was functioning well before this week. All domains are unblocked now. Many were blocked before.

* Operating system: Ubuntu 14.04 x86_64
* Chrome version: Version 42.0.2311.90 (64-bit)
* PB version: 2015.4.1

**Comment by cooperq** [comment]
@joepie91 @burgwinkel can you please send me a list of your other addons?

**Comment by burgwinkel** [comment]
Here's the list of my extensions from *chrome://inspect/#extensions*:

TransOver
Google+ Hangouts
Lazarus: Form Recovery
Awesome Screenshot: Screen capture, Annotate
WOT
BetaFish Adblocker
Service Pages for Google Chrome™
Web Developer
Slick RSS
Send from Gmail (by Google)
Session Buddy
Reverse Image Search
NotScripts for Chrome OS
FVD Video Downloader
CSS Shapes Editor
QR-Code Tag Extension
MEGA
RT News
AmazonSmile 1Button for Chrome
Privacy Badger
Bookmark Manager
Slick RSS : Feed Finder

**Comment by cooperq** [comment]
@joepie91 can you tell me what addons you have installed?
@burgwinkel if you disable other addons does privacy badger still not work?

**Comment by cooperq** [comment]
@mlane can you tell me which addons you have enabled?

**Comment by cooperq** [comment]
@poppyuser can you tell me which other addons you have enabled?

**Comment by dieseltravis** [comment]
This started happening for me on my chromebook after I updated recently. Here are the extensions enabled

* Advanced Font Settings 0.67
* Caret 1.5.9
* feedly 34
* GitHub EditorConfig 1.1.0
* Google Cast API 14.507.0.31277
* MightyText - SMS from PC & Text from Computer 15.5
* Privacy Badger 2015.4.1
* Secure Shell 0.8.34
* Calculator 2.1.0
* Camera 4.2.5
* Chrome Remote Desktop 42.0.2311.37
* Google Docs 0.9
* Google Keep - notes and lists 2.0.15262.1056
* Google Sheets 1.1
* Google Slides 0.9
* Google+ Photos 1.542.3
* Stylish 1.3.1
* VNC® Viewer for Google Chrome™ 1.2.0.5306

Chrome version info: 
Version 43.0.2357.130
Platform 6946.63.0 (Official Build) stable-channel daisy_spring
Firmware Google_Spring.3824.129.0

When inspecting the background page I see lots of messages like this in the console:
Error. Could not blacklist origin because no heuristic subscription found heuristicblocking.js:146

Occasionally errors like this:
![screenshot 2015-06-28 at 1 05 15 pm](https://cloud.githubusercontent.com/assets/17863/8397349/6962916c-1d96-11e5-9985-cd1ec3010f25.png)

**Comment by flexo3001** [comment]
same here. extentions enabled:
cVim
https everywhere
ublock origin
privacy badger

it makes no difference if extensions are enabled or not. I'm using chromium 43.0.2357.130 on linux and PB 2015.4.1.

**Comment by dieseltravis** [comment]
On my Windows 8 machine, Chrome version 43.0.2357.130 m (64-bit), with the exact same extensions enabled Privacy Badger 2015.4.1 still works fine.

**Comment by flexo3001** [comment]
in fact i think PB blocks but the count and sliders aren't working. 

![2015-06-29-203745_1366x768_scrot](https://cloud.githubusercontent.com/assets/1415596/8415432/f9921858-1e9e-11e5-9189-3779f9a6b6ae.png)

**Comment by cooperq** [comment]
ahhah! @dieseltravis I think your error is very telling:
`Error. Could not blacklist origin because no heuristic subscription found heuristicblocking.js:146`
This indicates that privacy badger has somehow had it's local storage data deleted. I will have to look further into how this could have happened.

**Comment by cooperq** [comment]
Do the rest of you see this error if you look at the privacy badger console? (you can go to the extensions page and check the box labeled "developer mode" and then click the `inspect views: background page` link to open the privacy badger console)

**Comment by flexo3001** [comment]
the only errors in my PB console
![2015-06-30-094238_1366x768_scrot](https://cloud.githubusercontent.com/assets/1415596/8426030/6e10478c-1f0c-11e5-957f-9143f31efd90.png)

**Comment by mlane3** [comment]
Well for me it might be fixed after I removed Scriptsafe like I said.  I was not able to check to see if "While I have not seen the "heuristic subscription error."; however when I added back scriptsafe and Google synced, it crashed privacy badger.  I wish I had know about the "background page" or else I would have recorded it for you.  
The extension I had enabled were:
Ghostery 5.4.3
Privacy Badger 2015.4.1
ScriptSafe 1.0.6.18

Needless to say, I have uninstalled my ad/popup blocker script safe, so I am going to have to find a new one soon.

Currently, the background page for privacy badger for me shows this:
![image](https://cloud.githubusercontent.com/assets/1785885/8431510/f6367f28-1f07-11e5-8cfe-44c876547f7b.png)

**Comment by SwartzCr** [comment]
@mlane3 I don't know if that code that is erroring out is still in Privacy Badger, can you click that link that says chrome-extension://lotsoftext/src/background.js:437:30 ? and show me what is says? That should show the line of the file that's giving the error

**Comment by SwartzCr** [comment]
@flexo3001 unfortunately I think those errors are irrelevant, let me know if you see any other errors though (I will also work to remove those errors)

**Comment by mlane3** [comment]
@SwartzCr 
Sure here is the pictures
![image](https://cloud.githubusercontent.com/assets/1785885/8639475/517656ba-28a9-11e5-9be3-bcb1dfcbeef6.png)
![image](https://cloud.githubusercontent.com/assets/1785885/8639483/9a791118-28a9-11e5-950a-8891358b1633.png)
Hope this helps

**Comment by mlane3** [comment]
So I assume from cooperq messages that its fixed?

**Comment by cooperq** [comment]
Sorry, it isn't fixed yet. Since we are having so much trouble reproducing it I removed it from the 1.0 milestone. Hopefully it will be fixed as a side effect of fixing other bugs when 1.0 comes out. If not we will have to keep researching it after the stable release.

**Comment by grant--** [comment]
I noticed that if you clear cache/cookies that it breaks the extension... perhaps it's related.

**Comment by pde** [comment]
@cooperq I am seeing this now, if you need a profile where it's reproduced.

**Comment by pde** [comment]
Diagnostics:
FilterStorage.knownSubscriptions is an empty Object
FilterStorage.subscriptions is an empty array

**Comment by cooperq** [comment]
Current prime suspect is the RemoveSubscription method is somehow getting called for all subscriptions. 
https://github.com/EFForg/privacybadgerchrome/blob/master/lib/adblockplus.js#L1338-L1355

**Comment by SwartzCr** [comment]
My guess is the load from disk function in adblock is getting called, failing to load, setting local variables as null, and they're never getting re-instituted by backround.js addSubscription, which runs on update and firstrun
adblock code is on line 1520, will continue to investigate

**Comment by cooperq** [comment]
Okay, This should be completely fixed in the latest version of privacy badger. As of today I have only deployed it to 10% of users (it has some huge changes!) but if you want to try it out early you can take the following steps. 
1. Download the crx file [from the chrome web store] (https://clients2.googleusercontent.com/crx/blobs/QgAAAC6zw0qH2DJtnXe8Z7rUJP1rX98iOE5SwxVXgdVTMTKqobiCFyOuS_jNFoVMdRhW-bEA7QhvVJw2hrKrZS8b8ThMYsv0t-wxxLr4D-Sm2ZC1AMZSmuW2zjq5MsW529CVI2Rtq2QO7xWEQg/pkehgijcmpdhfbdbbnkijodmdjhbjlgp_main.crx)
2. Open a new tab to [chrome://extensions](chrome://extensions)
3. Drag the downloaded file and drop it onto the extensions window. 
You should now have Privacy Badger version 2016.5.15
If this doesn't fix the problem please let me know. If you find any new bugs please file a separate issue.

**Comment by sveeke** [comment]
@cooperq the current version in the Chrome extention store and the version you are linking to both have the same problem on a laptop I installed Privacy Badger on. The buttons are all green and in the log there isn't any sign of blocking trackers. Do you (or someone else) have some pointers on how I can fix this?

**Comment by ghostwords** [comment]
Hi @sveeke, Privacy Badger doesn't block anything right away and needs to learn what to block first.

If you visit three or more newspaper sites, for example, Privacy Badger should learn to block a few domains. If you use other privacy/ad blocking extensions, Privacy Badger will take longer to learn, since most trackers will be blocked by your other extensions. You can test that Privacy Badger is working by temporarily disabling the other blockers, and visiting a few websites with just Privacy Badger.

Also note that  Privacy Badger does not learn in Incognito / Private Browsing windows.

Does this help?
